src/plugins/unified_search/public/search_bar/create_search_bar.tsx

```diff
@@ -218,7 +221,7 @@ export function createSearchBar({
             filters={filters}
             query={query}
             onFiltersUpdated={defaultFiltersUpdated(data.query, props.onFiltersUpdated)}
-            onRefreshChange={defaultOnRefreshChange(data.query)}
+            onRefreshChange={defaultOnRefreshChange(data.query, props.isAutoRefreshDisabled)}
```

Comment:
I think that there is no need to pass it to the function, just dont run the function if this prop is set to true

Reply:
I changed it to set `undefined` there if we have `isAutoRefreshDisabled` set to `true` otherwise it will use the function.

Reply:
I was thinking something like onRefreshChange={!props.isAutoRefreshDisabled ? defaultOnRefreshChange(data.query) : undefined}

Reply:
Oh I see, this is what you need. This is a nit but can you add the undefined as the else clause? As I have it on my comment?

Reply:
Sure, thanks for checking that! 🙂 I changed the condition to return the function if we don't have `isAutoRefreshDisabled` and otherwise to be `undefined`. Is that what you mean?